### ISSUE TITLE
[TextInputEditText/TextInputLayout] requestLayout() improperly called when prefixText is set on API 21

### ISSUE THREAD

**Issue by kuuuurt** [material-components/material-components-android#1673]
**Description:** My espresso tests are getting idle on an API 21 device with the following log
```
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{3a870d59 V.ED.... ......ID 0,0-57,164 #7f0901df app:id/textinput_prefix_text} during layout: running second layout pass
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatTextView{3a870d59 V.ED.... ........ 0,0-57,164 #7f0901df app:id/textinput_prefix_text} during second layout pass: posting in next frame
```

**Expected behavior:** It should improperly call requestLayout()

**Source code:** 
Just this in the XML file
```
<com.google.android.material.textfield.TextInputLayout
    android:id="@+id/til_phone_number"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    app:errorEnabled="true"
    app:prefixText="+">

                <com.google.android.material.textfield.TextInputEditText
                    android:id="@+id/edt_phone_number"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:hint="@string/phone_number_hint"
                    android:inputType="phone"
                    android:maxLength="12" />

</com.google.android.material.textfield.TextInputLayout>
```

**Android API version:** 21

**Material Library version:** 1.3.0-alpha02

**Device:** Emulator using API 21

**Comment by hram** [comment]
Any progress for this issue?
catalog-debug.apk reprodused this issue

Also this issue blocks espresso UI tests

**Comment by Protino** [comment]
This is occurring on API 22 with Material Library version: 1.3.0 as well.

**Comment by Protino** [comment]
Looks like this has been fixed in 1.4.0, please confirm and include it in the release notes if so.

**Comment by drchen** [comment]
I believe it's a framework issue of `TextView` on Lollipop (API 21 & 22). 

I agree it's a really bad issue but it happens in a lot of different cases and we really have no good way to have a perfect solution on our end to fix them. It's unfortunate but we probably need to skip espresso UI tests for this case on Lollipop.
